doc/contributing.md

```diff
@@ -1,16 +1,171 @@
+
+```bash
+conda install -c pyviz/label/dev "pyctdev>0.5.0"
+```
+
+Once `pyctdev` is available and you are in the cloned panel repository, you can create a new virtual environment with the `env_create` *doit* command with the appropriate channels:
+
+```bash
+doit env_create -c channel1 -c channel2 --name=dev-env --python=3.x
+```
+
+Don't forget to activate this environment:
+
+```bash
+conda activate dev-env
+```
+
+To perform an editable install of the project you are working on, including all the dependencies required to run the full unit test suite, run the `develop_install` with the appropriate channels and options:
+
+```bash
+doit develop_install -c channel1 -c channel2 -o tests -o ... -o ...
+```
+
+Depending on the options you have chosen to install (usually `-o tests` for the unit tests, `-o examples` to run the examples notebooks, ...), the following commands will run:
+
+* The unit tests: `doit test_unit.`
+* The examples tests: `doit test_examples.`
+
+```{warning}
+The *options* and *commands* are not standardized across the HoloViz projects. Refer to the project documentation for the exact commands to run.
+```
+
+```{hint}
+The reference sources to find out which commands to run to install the correct dependencies and to execute the tests and other tasks are the **Github Actions Workflows** files that you can find in the `.github` folder of each repository. 
+```
+
+At this step, you should have your environment set up, being able to run tests and a clone of the source repository. The next step is to create a branch and start making changes. Before committing these changes, make sure the tests still pass by running them. The HoloViz source codes are stringent concerning styling (e.g., they're not using [black](https://github.com/psf/black)). Try to write code that follows the style of the surrounding code. Once you are done with your changes, you can commit them. It is good practice to break down big changes into smaller chunks, and each chunk has its own commit with a message that summarizes it. Now you can push the branch to your clone and create a *pull request* from your clone to the original repository. This [Github Gist](https://gist.github.com/Chaser324/ce0505fbed06b947d962) describes these GitHub steps in more detail.
+
+The *pull request* you make should reference the *issue* you are attempting to close (i.e. `Fixes #issuenumber`) and include a description of the changes you made. Changes affecting the visual properties should include screenshots or GIFs.
+
+Your *pull request* should now be reviewed by one of the project's maintainers. This may take a while given how busy the maintainers are, so try to be patient :) Once the review is done you may be required to make some changes. You may also be asked to *resolve conflicts* if the changes you made appear to conflict with changes made to the source code after you submitted the *pull request*.
```

Comment:
```suggestion
Your *pull request* should now be reviewed by one of the project's maintainers. This may take a while given how busy the maintainers are, so try to be patient :). Your pull request will be evaluated to ensure that it is within the scope of the project (again, it's a good idea to create an issue first to outline your intent and give the maintainers an opportunity to weigh in before you spend the effort creating a pull request). If it's decided to proceed, the review will be conducted, and once the review is done you may be required to make some changes. You may also be asked to *resolve conflicts* if the changes you made appear to conflict with changes made to the source code after you submitted the *pull request*.
```